From b299116b38da458213e3b74559e1997165d1ff72 Mon Sep 17 00:00:00 2001
From: Lu Yahan <yahan@iscas.ac.cn>
Date: Mon, 19 Jun 2023 10:11:37 +0800
Subject: [PATCH] [riscv][SFI] Fix the store size of SharedFunctionInfo::Age
 field

The age field has an uint16 type so it should be "store half".


Change-Id: I5a831d1d9655697ac062381cb38bf396a0bbddf6
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/4624977
Reviewed-by: Ji Qiu <qiuji@iscas.ac.cn>
Commit-Queue: Yahan Lu <yahan@iscas.ac.cn>
Commit-Queue: Ji Qiu <qiuji@iscas.ac.cn>
Auto-Submit: Yahan Lu <yahan@iscas.ac.cn>
Cr-Commit-Position: refs/heads/main@{#88336}
---
 src/builtins/riscv/builtins-riscv.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/builtins/riscv/builtins-riscv.cc b/src/builtins/riscv/builtins-riscv.cc
index e0dc7194ba9..81509a57763 100644
--- a/src/builtins/riscv/builtins-riscv.cc
+++ b/src/builtins/riscv/builtins-riscv.cc
@@ -943,7 +943,7 @@ static void AdvanceBytecodeOffsetOrReturn(MacroAssembler* masm,
 
 namespace {
 void ResetSharedFunctionInfoAge(MacroAssembler* masm, Register sfi) {
-  __ StoreWord(zero_reg, FieldMemOperand(sfi, SharedFunctionInfo::kAgeOffset));
+  __ Sh(zero_reg, FieldMemOperand(sfi, SharedFunctionInfo::kAgeOffset));
 }
 
 void ResetJSFunctionAge(MacroAssembler* masm, Register js_function,
-- 
2.35.1

